### PR TITLE
added some docstrings to spatial data

### PR DESCRIPTION
**Pull request by EliHei2**
Not even close to complete, but just making the PR for things to be able to move forward!

### CODE REVIEW

File: spatialdata/_core/_spatialdata.py [hunk 1]
```diff
@@ -200,6 +244,20 @@ def is_backed(self) -> bool:
 
     # TODO: from a commennt from Giovanni: consolite somewhere in a future PR (luca: also _init_add_element could be cleaned)
     def _get_group_for_element(self, name: str, element_type: str) -> zarr.Group:
+        """Get the group for an elemnt, creates a new one if the element doesn't exist
+
+        Parameters
+        ----------
+        name : str
```
Comment (giovp) [inline spatialdata/_core/_spatialdata.py:251]: in general, parameters should not have types as they are already present in signature. so this (and everything else) should just be `name`

File: spatialdata/_core/_spatialdata.py [hunk 2]
```diff
@@ -200,6 +244,20 @@ def is_backed(self) -> bool:
...
+            name of the element
+        element_type : str
+            type of the element. Should be in ["images", "labels", "points", "polygons", "shapes"].
+
+        Returns
+        -------
+        zarr.Group
+            either the exisitng sub-group or a new one
```
Comment (giovp) [inline spatialdata/_core/_spatialdata.py:259]: same here, this one should just be
`:func:`zarr.hierarchy.group` following the definition in e.g. [here](https://zarr.readthedocs.io/en/stable/api/hierarchy.html)

**Review by giovp — CHANGES_REQUESTED** [review]
hi @EliHei2 ,

thanks for adding docstrings to spatialdata, I left couple of comments that should be implemented in the rest of docstrings, unfortunately we do it slightly different than what you suggested (numpy style). It's usually a good practice to first take a look at how docs are defined for already existing functions/classes and copy from there.

thanks again for starting this!

**Comment by LucaMarconato** [comment]
One more thing, please remove the file `spatialdata/_io/.ipynb_checkpoints/read-checkpoint.py` from the pr.

**Pull request closed without merging**